### PR TITLE
Dev 94318 react native webview example app

### PR DESCRIPTION
**Pull request by marcosrivereto**

### CODE REVIEW

File: ReactNativeApp/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "ReactNativeApp",
```
Comment (lukevance) [inline ReactNativeApp/package.json:2]: Whatever name we choose for the app (trustly-react-native-example) should be here too

File: ReactNativeApp/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+import { shouldOpenInAppBrowser } from "./oauthUtils";
+
+import { InAppBrowser } from 'react-native-inappbrowser-reborn'
+import {Colors} from 'react-native/Libraries/NewAppScreen';
+
+export default class App extends Component {
+  trustlyWebView = null;
+
+  widgetData = {
+    accessId: "A48B73F694C4C8EE6306",
```
Comment (lukevance) [inline ReactNativeApp/src/App.tsx:20]: replace with "YOUR_ACCESS_ID"

File: ReactNativeApp/src/trustly.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+export const widget = (data = {}) => `<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Trustly</title>
+  <script src="http://192.168.0.9:8000/start/scripts/pwmb.js?accessId=A48B73F694C4C8EE6306" type="text/javascript"></script>
```
Comment (lukevance) [inline ReactNativeApp/src/trustly.tsx:7]: update base URL to `sandbox.trustly.com` and update `accessId` to "YOUR_ACCESS_ID"

File: react-web-app/public/index.html [hunk 4]
```diff
@@ -24,7 +24,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <script src="https://sandbox.trustly.one/start/scripts/trustly.js?accessId={YOUR_ACCESS_ID}"></script>
+    <script src="https://sandbox.trustly.one/start/scripts/trustly.js?accessId=A48B73F694C4C8EE6306"></script>
```
Comment (lukevance) [inline react-web-app/public/index.html:27]: please revert this change

**Review by lukevance — CHANGES_REQUESTED** [review]
After review this further, I do think that it will be too confusing for viewers to understand that this project contains a React web app and also a React Native app. But if this is too much of a hassle, please just address the comments and I can move the project to this repository: https://github.com/TrustlyInc/trustly-react-native-example

File: ReactNativeApp/app.json
[the remark this answers is not in the document]
Comment (marcosrivereto) [reply ReactNativeApp/app.json:1]: Changed to TrustlyReactNativeExample, I don't know why react-native cli is not accepting names with "-".

File: ReactNativeApp/src/App.tsx [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    `;
+
+    return (
+
+        <SafeAreaView style={backgroundStyle}>
+          <WebView
+              ref={(ref) => (this.trustlyWebView = ref)}
+              source={{ html: widget(this.widgetData) }}
+              renderLoading={this.LoadingIndicatorView}
+              onNavigationStateChange={this.handleWidgetStateChange}
```
[the remark this answers is not in the document]
Comment (marcosrivereto) [reply ReactNativeApp/src/App.tsx:164]: No, we can remove this part.

File: ReactNativeApp/src/App.tsx [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+      <ActivityIndicator color="#333" size="small" style={styles.loading} />
+    );
+  }
+
+
+  handleWidgetStateChange = (newNavState) => {
+    console.log("handleWidgetStateChange: ", newNavState);
+    const { url } = newNavState;
+    if (!url) return;
+
```
[the remark this answers is not in the document]
Comment (marcosrivereto) [reply ReactNativeApp/src/App.tsx:60]: No, we can remove this part.

File: ReactNativeApp/src/trustly.tsx (hunk 3 above)
Comment (marcosrivereto) [reply ReactNativeApp/src/trustly.tsx:7]: As we moved the React Native to trustly-react-native-example repo, we can just ignore this PR and consider the new one. New PR https://github.com/TrustlyInc/trustly-react-native-example/pull/1

File: react-web-app/src/App.js [hunk 7]
```diff
@@ -1,8 +1,9 @@
 import PayCard from './PayCard';
 
-const ACCESS_ID = 'YOUR_ACCESS_ID';
-const MERCHANT_ID = 'YOUR_MERCHANT_ID';
-const serverURL = null; // YOUR_SERVER_URL
+const ACCESS_ID = 'A48B73F694C4C8EE6306';
+const MERCHANT_ID = '110005514';
+const serverURL = 'https://sandbox.paywithmybank.com'; // YOUR_SERVER_URL
```
[the remark this answers is not in the document]
Comment (marcosrivereto) [reply react-web-app/src/App.js:5]: As we moved the React Native to trustly-react-native-example repo, we can just ignore this PR and consider the new one. New PR https://github.com/TrustlyInc/trustly-react-native-example/pull/1

File: react-web-app/public/index.html (hunk 4 above)
Comment (marcosrivereto) [reply react-web-app/public/index.html:27]: As we moved the React Native to trustly-react-native-example repo, we can just ignore this PR and consider the new one. New PR https://github.com/TrustlyInc/trustly-react-native-example/pull/1
